components/x-gift-article/src/lib/api.js

```diff
@@ -0,0 +1,63 @@
+const REDEMPTION_LIMIT = 3;
+
+module.exports = {
+	getGiftArticleAllowance: async () => {
+		try {
+			const response = await fetch('/article-email/credits', { credentials: 'same-origin' });
+			const json = await response.json();
+
+			return {
+				monthlyAllowance: json.credits.allowance,
+				giftCredits: json.credits.remainingCredits
+			};
+
+		} catch (e) {
+			return { monthlyAllowance: undefined, giftCredits: undefined };
+		}
+
+	},
+
+	getGiftUrl: async (articleId, sessionId) => {
+		try {
+			const response = await fetch('/article-email/gift-link', {
+				credentials: 'same-origin',
+				method: 'POST',
+				headers: { 'Content-Type': 'application/json' },
+				body: JSON.stringify({
+					contentUUID: articleId,
+					ftSessionSecure: sessionId
+				})
+			});
+
+			const body = response.ok ? await response.json() : {};
+			if (body.errors) {
+				throw new Error(`Failed to get gift article link: ${body.errors.join(', ')}`);
+			}
+
+			return Object.assign({}, body, { redemptionLimit: REDEMPTION_LIMIT });
```

Comment:
You could use the shorter [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals) here: 

```js
return {
  ...body,
  redemptionLimit: REDEMPTION_LIMIT
}
```

Reply:
Oh, I see! thank you!!:simple_smile: